src/libraries/System.Private.CoreLib/src/System/IO/StreamReader.cs

```diff
@@ -786,43 +788,39 @@ private int ReadBuffer(Span<char> userBuffer, out bool readToUserBuffer)
                 }
             }
 
-            StringBuilder? sb = null;
+            using ValueStringBuilder sb = new(stackalloc char[256]);
```

Comment:
This code is essentially the same as in [StringReader.ReadLine](https://github.com/danmoseley/runtime/blob/78bf14a4111a0b6760a2321b6f8759c83674b216/src/libraries/System.Private.CoreLib/src/System/IO/StringReader.cs#L178-L217)

Should they have the same implementation? (Or even possibly shared and force inlined)

Reply:
Yeah, they look almost the same but still I would stick this PR to stream reader only :)